app/bundles/CampaignBundle/Event/EventSchedulerCalculationEvent.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/*
+ * @copyright   2018 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\CampaignBundle\Event;
+
+use Mautic\LeadBundle\Entity\Lead;
+use Symfony\Component\EventDispatcher\Event;
+
+/**
+ * Class EventSchedulerCalculationEvent.
+ */
+class EventSchedulerCalculationEvent extends Event
+{
+    /**
+     * @var \Mautic\CampaignBundle\Entity\Event
+     */
+    private $event;
+
+    /**
+     * @var \DateTime
+     */
+    private $compareFromDateTime;
+
+    /**
+     * @var \DateTime
+     */
+    private $comparedToDateTime;
+
+    /**
+     * @var Lead
+     */
+    private $contact;
+
+    /**
+     * @var \DateTime
+     */
+    private $executionDateTime;
+
+    /**
+     * EventSchedulerCalculationEvent constructor.
+     *
+     * @param Event     $event
+     * @param DateTime  $compareFromDateTime
+     * @param DateTime  $comparedToDateTime
+     * @param Lead|null $contact
+     * @param DateTime  $executionDateTime
+     */
+    public function __construct(\Mautic\CampaignBundle\Entity\Event $event, \DateTime $compareFromDateTime, \DateTime $comparedToDateTime, Lead $contact = null, \DateTime $executionDateTime)
```

Comment:
Can we use `\DateTimeInterface` instead of `\DateTime` in all occurrences in this PR?

Reply:
Also there is git conflict. Please merge current staging into your branch and resolve it.